pyccel/ast/operators.py

```diff
@@ -70,11 +73,20 @@ def broadcast(shape_1, shape_2):
 
     new_shape = []
     for e1,e2 in zip(new_shape_1, new_shape_2):
-        if e1 == e2:
+        if pyccel_to_sympy(e1, {}, set()) == pyccel_to_sympy(e2, {}, set()):
+            new_shape.append(e1)
+        elif pyccel_to_sympy(e1, {}, set()) == 1:
+            new_shape.append(e2)
+        elif pyccel_to_sympy(e2, {}, set()) == 1:
+            new_shape.append(e1)
+        elif pyccel_to_sympy(e1, {}, set()).is_constant() and not\
+                pyccel_to_sympy(e2, {}, set()).is_constant():
```

Comment:
can you just create two variables to store the result of pyccel_to_sympy? and use those two variables in the conditions